### PR TITLE
LPS-199294 & LPS-199183 Add POSHI automation | Master 

### PR DESCRIPTION
**Pull request by alessandromalves**
This is only for master
https://liferay.atlassian.net/browse/LPS-199294
https://liferay.atlassian.net/browse/LPS-199183

cc.: @david-gutierrez-mesa @Tim-Cao

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by alessandromalves** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplatesForObject.testcase [hunk 1]
```diff
@@ -1205,4 +1205,207 @@ definition {
...
+			AssertElementPresent(
+				key_text = "title",
+				locator1 = "CKEditor#BODY_TEXT");
+		}
+	}
+
+	@description = "This is a test for LPS-199013. View Select From List fragment display success in a Display Page when entry field is empty."
+	@priority = 5
+	test ViewSelectFromListFragmentDisplaySuccessWithEntryFieldEmpty {
+		property custom.properties = "feature.flag.LPS-181663=true${line.separator}feature.flag.LPS-183727=true${line.separator}feature.flag.LPS-195205=true";
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplatesForObject.testcase:1313]: Same as above

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplatesForObject.testcase [hunk 2]
```diff
@@ -1205,4 +1205,207 @@ definition {
...
+					picklistName = "Fruits");
+			}
+
+			ObjectAdmin.addObjectFieldViaAPI(
+				fieldBusinessType = "RichText",
+				fieldLabelName = "Title",
+				fieldName = "title",
+				fieldType = "Clob",
+				isRequired = "false",
+				objectName = "PurchaseOrder");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplatesForObject.testcase:1331]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplatesForObject.testcase [hunk 3]
```diff
@@ -1205,4 +1205,207 @@ definition {
...
+
+			PageEditor.viewInputStyle(
+				label = "Title",
+				type = "rich-text-input");
+
+			SelectFrame(locator1 = "IFrame#IFRAME");
+
+			AssertElementPresent(
+				key_text = "title",
+				locator1 = "CKEditor#BODY_TEXT");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplatesForObject.testcase:1407]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplatesForObject.testcase [hunk 4]
```diff
@@ -1205,4 +1205,207 @@ definition {
...
+		}
+
+		task ("When a new submission is created with a entry field empty") {
+			ContentPagesNavigator.openViewContentPage(
+				pageName = "Test Page Name",
+				siteName = ${randomSiteName});
+
+			FormFields.addRichTextContent(
+				content = "title",
+				fieldLabel = "Title");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplatesForObject.testcase:1389]: No need

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplatesForObject.testcase [hunk 5]
```diff
@@ -1205,4 +1205,207 @@ definition {
...
+
+			Button.clickSubmitButton();
+		}
+
+		task ("Then the Success Interaction will be completed") {
+			PageEditor.viewInputStyle(
+				placeholder = "Choose an Option",
+				type = "select-from-list");
+
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplatesForObject.testcase:1398]: Add step to assert redirect to the display page

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplatesForObject.testcase [hunk 6]
```diff
@@ -1205,4 +1205,207 @@ definition {
 		}
 	}
 
+	@description = "This is a test for LPS-198511. View Multiselect List fragment display success in a Display Page when entry field is empty."
+	@priority = 5
+	test ViewMultiselectListFragmentDisplaySuccessWithEntryFieldEmpty {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplatesForObject.testcase:1210]: ViewMultiselectListWithEntryFieldEmptyOnAssociatedDisplayPageTemplate

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplatesForObject.testcase [hunk 7]
```diff
@@ -1205,4 +1205,207 @@ definition {
...
+
+			AssertElementPresent(
+				key_text = "title",
+				locator1 = "CKEditor#BODY_TEXT");
+		}
+	}
+
+	@description = "This is a test for LPS-199013. View Select From List fragment display success in a Display Page when entry field is empty."
+	@priority = 5
+	test ViewSelectFromListFragmentDisplaySuccessWithEntryFieldEmpty {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/layout/pagetemplate/displaypagetemplate/DisplayPageTemplatesForObject.testcase:1312]: ViewSelectFromListWithEntryFieldEmptyOnAssociatedDisplayPageTemplate

**Review by Tim-Cao — CHANGES_REQUESTED** [review]
Hi @alessandromalves, could you update according to comments? Thanks

**Comment by alessandromalves** [comment]
ci;reopen

**Comment by alessandromalves** [comment]
ci:reopen

**Comment by alessandromalves** [comment]
Hi @Tim-Cao, I needed to update `JSONObject.getObjectEntryId macro` to make it work when `entryValue` is empty. However, even with the change, it didn't work when the field is an select-from-list. Do you know how I can fix this? Thanks in advance!

**Comment by alessandromalves** [comment]
ci:test:sf

**Comment by Tim-Cao** [comment]
@alessandromalves, see changes https://github.com/liferay-echo/liferay-portal/pull/14466

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**